airflow/providers/cncf/kubernetes/hooks/kubernetes_hook.py

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import tempfile
+
+from kubernetes import client, config
+
+from airflow.hooks.base_hook import BaseHook
+
+
+class Kuberneteshook(BaseHook):
```

Comment:
I wonder if we could share some of the connection code we already have for both the KubernetesPodOperator and the Kubernetes Executor. Or should the KubePodOperator be slowly reworked (in a separate PR) to use this hook?

Reply:
I feel like KubePodOperator should use this hook.

Reply:
Should be a separate PR though

Reply:
so would you like me to open new PR for the Kubernetes connection? I think that KubePodOperator can benefit from using this hook as it can make it work on remote k8s clusters.

Reply:
Non, this PR can stay as it is, just talking about future work.

KubePodOp already works against external clusters though